packages/documentation/src/stories/components/collapsible/collapsible.docs.mdx

```diff
@@ -0,0 +1,70 @@
+import { Meta, Canvas, Story, Source, ArgsTable } from '@storybook/addon-docs';
+
+<Meta title="Components/Collapsible" />
+
+# Collapsible
+
+**Toggle the visibility of content across your project.**<br/>
+The `<post-collapsible>` component is used to show and hide content.
+Collapsing an element will animate the height from its current value to 0.
+
+<Canvas>
+  <Story id="components-collapsible--default" />
+</Canvas>
+
+<ArgsTable story="^" />
+
+## Examples
+
+The following examples show different use cases for the `<post-collapsible>` component.
+
+### Initially Collapsed
+
+To make the collapsible content hidden by default, just use the `collapsible="true"` property.
+
+<Canvas>
+  <Story id="components-collapsible--initially-collapsed" />
+</Canvas>
+
+### Heading Level
+
+Use the `heading-level` property to define the hierarchical level of the collapsible header within the structure of a document.
+
+<Canvas>
+  <Story id="components-collapsible--heading-level" />
+</Canvas>
+
+### Intricate Content
+
+The collapsible component uses [HTML slot elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot)
+to allow any HTML content to be used as part of its header and body.<br/>
+By default, all children are appended to the component's body.
+For a specific child to be displayed in the header, a `slot="header"` attribute must be added.
+
+
+<Canvas withSource>
+  <Story id="components-collapsible--intricate-content" />
+</Canvas>
+
+### Custom Trigger
+
+The `<post-collapsible>` component exposes a `.toggle()` method that allows to trigger the collapse programmatically.
+This method is asynchronous and returns a promise that resolves with the current open state.
+It optionally takes a boolean parameter that forces open when `true` or close when `false`.
+
+```typescript dark
+const collapsible = document.querySelector('#collapsibleId') as HTMLPostCollapsibleElement;
+collapsible.toggle(/* open: boolean */).then((/* isOpen: boolean */) => {});
+```
```

Comment:
Maybe we should also provide a non-typescript example.
What do you think?

Reply:
Do you mean having both TS and JS or having JS only?

The JS version is very similar to the TS except it gives extra information with `as HTMLPostCollapsibleElement`:
```suggestion
```typescript dark
const collapsible = document.querySelector('#collapsibleId');
collapsible.toggle(/* open: boolean */).then((/* isOpen: boolean */) => {});
```
```